### ISSUE TITLE
CI testing should include a Release build

### ISSUE THREAD

**Issue by shamilovtim** [facebook/flipper#3043]
I was curious how compile time errors have been making it into the rep so I checked out what CI tests and found that at least on iOS, CI builds with a Debug configuration
<img width="1780" alt="Screen Shot 2021-11-11 at 8 11 00 PM" src="https://user-images.githubusercontent.com/3495974/141391710-fe1380b0-9619-48bc-a434-ed888065c59f.png">

If you read bug reports users are complaining about Release builds failing, while maintainers are telling users that they are not able to reproduce such errors. I realize that part of the issue is that Flipper is supposed to be excluded from Release builds, but the problem is that Flipper stills attempts to build in Release mode on React Native. Users still complain about Release errors, but because the CI steps do not even attempt to create a Release build, maintainers cannot reproduce it. 

I suggest adding a Release build to the CI steps for at least iOS

**Comment by passy** [comment]
Yeah, that sounds reasonable. We've had some problems with capacity before because these builds take quite a long time and use up a good chunk of CPU, but it's worth a try.
